perceval/runtime/job_group.py

```diff
@@ -0,0 +1,281 @@
+        Adds information of the new RemoteJob to an existing Group.
+        Saves the data in a chronological order in the group (each entry is
+        a dictionary of necessary information - status, id, body, metadata)
+        """
+        if not isinstance(job_to_add, RemoteJob):
+            raise TypeError('Only a RemoteJob can be added to the group')
+
+        job_info = dict()
+        job_info['id'] = job_to_add.id
+
+        if job_to_add.id is None:
+            # set status to None for Jobs not sent to cloud
+            job_info['status'] = None
+            # Save information inside body (circuit, payload, etc.) to send job later
+            job_info['body'] = job_to_add._request_data
+            max_samples = kwargs.get('max_samples', MAX_SAMPLE_DEF_GROUP)
+            job_info['args'] = {"max_samples": max_samples}
+        else:
+            job_info['status'] = job_to_add.status()
+
+        if not job_info['status'] == 'SUCCESS':
+            # save metadata for any job that did not run successfully or was not sent
+            job_info['metadata'] = {'headers': job_to_add._rpc_handler.headers,
+                                    'platform': job_to_add._rpc_handler.name,
+                                    'url': job_to_add._rpc_handler.url}
+
+        self._group_info['job_group_data'].append(job_info)  # save changes in object
+        # include the added job's info to dataset in the group
+        self._modify_job_dataset(job_info)
+
+    def _modify_job_dataset(self, updated_info: dict):
+        # modifies the recorded information when a new job is added
+        group_data = self._read_job_group_from_disk()
+
+        saved_job_info = group_data['job_group_data']
+        saved_job_info.append(updated_info)
+
+        # self._group_info['job_group_data'] = saved_job_info  # save changes in object
+        self._write_job_group_to_disk()  # save changes to disk
+
+    def progress(self):
+        """
+        Lists through all jobs in the group and returns a list of status
+        If a status is changed from existing in file, that entry is
+        updated with new information
+        """
+        status_jobs_in_group = []
+
+        for job_entry in self._group_info['job_group_data']:
+            if not job_entry['status'] in ['SUCCESS', None]:
+                rj = self._recreate_remote_job_from_stored_data(job_entry)
+                rj_status = rj.status()
+                if rj_status == 'SUCCESS':
+                    # remove metadata information if job ran successfully
+                    del job_entry['metadata']
+                job_entry['status'] = rj_status  # update current status
+
+            status_jobs_in_group.append(job_entry['status'])
+
+        self._write_job_group_to_disk()  # rewrites the group information on disk
+
+        # replace None with 'Not_sent' - more readable
+        status_list = [status if status is not None else "Not_sent" for status in status_jobs_in_group]
```

Comment:
What are users going to do with a list of statuses? We can answer this question in the other ticket, but right now I guess the progress() return value isn't really usable.

Reply:
Yes, in this state it is not usable. The idea of having this method here is simply to have the structure of the method ready, ii.e. how it behaves for each job in the group. Do you want me to update it to not return anything here? Or is it ok if we leave it and finish the `progress` method in the next ticket?